addons/account_edi_ubl_cii/models/ir_actions_report.py

```diff
@@ -62,4 +68,37 @@ def _render_qweb_pdf_prepare_streams(self, report_ref, data, res_ids=None):
                 invoice = self.env['account.move'].browse(res_id)
                 self._add_pdf_into_invoice_xml(invoice, stream_data)
 
+            # If Factur-X isn't already generated, generate and embed it inside the PDF
+            if len(res_ids) == 1:
+                invoice = self.env['account.move'].browse(res_ids)
+                if invoice.is_sale_document() and invoice.state != 'draft':
+                    # Add the attachments to the pdf file
+                    pdf_stream = collected_streams[invoice.id]['stream']
+
+                    # Read pdf content.
+                    pdf_content = pdf_stream.getvalue()
+                    reader_buffer = io.BytesIO(pdf_content)
+                    reader = OdooPdfFileReader(reader_buffer, strict=False)
+
+                    # Post-process and embed the additional files.
+                    writer = OdooPdfFileWriter()
+                    writer.cloneReaderDocumentRoot(reader)
+
+                    # If Factur-X hasn't been generated, generate and embed it anyway !
+                    if 'facturx_1_0_05' not in invoice.edi_document_ids.edi_format_id.mapped('code'):
```

Comment:
> You also need to test the invoice is posted I think

I don't think so, in account_edi, we only test `invoice.state != 'draft'` before embedding additional files using `_prepare_invoice_report`


> ?

Then, if NLCIUS (for instance) is enabled but not FacturX, FacturX will not be embedded in the PDF.

> you need to check this before making all that stuff. Otherwise, you are making all of it for nothing

Indeed, oops